{
  "packageName": "java.sql",
  "simpleName": "SQLException",
  "moduleName": "java.sql",
  "type": "CLASS",
  "superClass": {
    "name": "java.lang.Exception",
    "type": "Class"
  },
  "members": [
    {
      "returnType": {
        "name": "java.lang.String",
        "type": "Class"
      },
      "parameters": [],
      "memberType": "JavaMethod",
      "modifiers": 1,
      "name": "getSQLState",
      "comment": "Retrieves the SQLState for this SQLException object.",
      "tagMap": {
        "return": [
          "the SQLState value"
        ]
      }
    },
    {
      "returnType": {
        "name": "int",
        "type": "Class"
      },
      "parameters": [],
      "memberType": "JavaMethod",
      "modifiers": 1,
      "name": "getErrorCode",
      "comment": "Retrieves the vendor-specific exception code for this SQLException object.",
      "tagMap": {
        "return": [
          "the vendor\u0027s error code"
        ]
      }
    },
    {
      "returnType": {
        "name": "java.sql.SQLException",
        "type": "Class"
      },
      "parameters": [],
      "memberType": "JavaMethod",
      "modifiers": 1,
      "name": "getNextException",
      "comment": "Retrieves the exception chained to this SQLException object by setNextException(SQLException ex).",
      "tagMap": {
        "see": [
          "\u003ca href\u003d\"https://docs.oracle.com/en/java/javase/16/docs/api/java.sql/java/sql/SQLException.html#setNextException(java.sql.SQLException)\"\u003e\u003ccode\u003esetNextException(java.sql.SQLException)\u003c/code\u003e\u003c/a\u003e"
        ],
        "return": [
          "the next \n\u003ccode\u003eSQLException\u003c/code\u003e object in the chain; \n\u003ccode\u003enull\u003c/code\u003e if there are none"
        ]
      }
    },
    {
      "returnType": {
        "name": "void",
        "type": "Class"
      },
      "parameters": [
        {
          "type": {
            "name": "java.sql.SQLException",
            "type": "Class"
          },
          "field": {
            "name": "ex"
          }
        }
      ],
      "memberType": "JavaMethod",
      "modifiers": 1,
      "name": "setNextException",
      "comment": "Adds an SQLException object to the end of the chain.",
      "tagMap": {
        "see": [
          "\u003ca href\u003d\"https://docs.oracle.com/en/java/javase/16/docs/api/java.sql/java/sql/SQLException.html#getNextException()\"\u003e\u003ccode\u003egetNextException()\u003c/code\u003e\u003c/a\u003e"
        ],
        "param": [
          "\u003ccode\u003eex\u003c/code\u003e - the new exception that will be added to the end of the \n\u003ccode\u003eSQLException\u003c/code\u003e chain"
        ]
      }
    },
    {
      "returnType": {
        "name": "java.util.Iterator",
        "parameters": [
          {
            "name": "java.lang.Throwable",
            "type": "Class"
          }
        ],
        "type": "Class"
      },
      "parameters": [],
      "memberType": "JavaMethod",
      "modifiers": 1,
      "name": "iterator",
      "comment": "Returns an iterator over the chained SQLExceptions. The iterator will be used to iterate over each SQLException and its underlying cause (if any).",
      "tagMap": {
        "Specified by:": [
          "\u003ccode\u003e\u003ca href\u003d\"https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/lang/Iterable.html#iterator()\"\u003eiterator\u003c/a\u003e\u003c/code\u003e\u0026nbsp;in interface\u0026nbsp;\n\u003ccode\u003e\u003ca href\u003d\"https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/lang/Iterable.html\" title\u003d\"interface in java.lang\"\u003eIterable\u003c/a\u003e\u0026lt;\u003ca href\u003d\"https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/lang/Throwable.html\" title\u003d\"class in java.lang\"\u003eThrowable\u003c/a\u003e\u0026gt;\u003c/code\u003e"
        ],
        "return": [
          "an iterator over the chained SQLExceptions and causes in the proper order"
        ],
        "since": [
          "1.6"
        ]
      }
    },
    {
      "parameters": [
        {
          "type": {
            "name": "java.lang.String",
            "type": "Class"
          },
          "field": {
            "name": "reason"
          }
        },
        {
          "type": {
            "name": "java.lang.String",
            "type": "Class"
          },
          "field": {
            "name": "SQLState"
          }
        },
        {
          "type": {
            "name": "int",
            "type": "Class"
          },
          "field": {
            "name": "vendorCode"
          }
        }
      ],
      "memberType": "JavaMethod",
      "modifiers": 1,
      "comment": "Constructs a SQLException object with a given reason, SQLState and vendorCode. The cause is not initialized, and may subsequently be initialized by a call to the Throwable.initCause(java.lang.Throwable) method.",
      "tagMap": {
        "param": [
          "\u003ccode\u003ereason\u003c/code\u003e - a description of the exception",
          "\u003ccode\u003eSQLState\u003c/code\u003e - an XOPEN or SQL:2003 code identifying the exception",
          "\u003ccode\u003evendorCode\u003c/code\u003e - a database vendor-specific exception code"
        ]
      }
    },
    {
      "parameters": [
        {
          "type": {
            "name": "java.lang.String",
            "type": "Class"
          },
          "field": {
            "name": "reason"
          }
        },
        {
          "type": {
            "name": "java.lang.String",
            "type": "Class"
          },
          "field": {
            "name": "SQLState"
          }
        }
      ],
      "memberType": "JavaMethod",
      "modifiers": 1,
      "comment": "Constructs a SQLException object with a given reason and SQLState. The cause is not initialized, and may subsequently be initialized by a call to the Throwable.initCause(java.lang.Throwable) method. The vendor code is initialized to 0.",
      "tagMap": {
        "param": [
          "\u003ccode\u003ereason\u003c/code\u003e - a description of the exception",
          "\u003ccode\u003eSQLState\u003c/code\u003e - an XOPEN or SQL:2003 code identifying the exception"
        ]
      }
    },
    {
      "parameters": [
        {
          "type": {
            "name": "java.lang.String",
            "type": "Class"
          },
          "field": {
            "name": "reason"
          }
        }
      ],
      "memberType": "JavaMethod",
      "modifiers": 1,
      "comment": "Constructs a SQLException object with a given reason. The SQLState is initialized to null and the vendor code is initialized to 0. The cause is not initialized, and may subsequently be initialized by a call to the Throwable.initCause(java.lang.Throwable) method.",
      "tagMap": {
        "param": [
          "\u003ccode\u003ereason\u003c/code\u003e - a description of the exception"
        ]
      }
    },
    {
      "parameters": [],
      "memberType": "JavaMethod",
      "modifiers": 1,
      "comment": "Constructs a SQLException object. The reason, SQLState are initialized to null and the vendor code is initialized to 0. The cause is not initialized, and may subsequently be initialized by a call to the Throwable.initCause(java.lang.Throwable) method.",
      "tagMap": {}
    },
    {
      "parameters": [
        {
          "type": {
            "name": "java.lang.Throwable",
            "type": "Class"
          },
          "field": {
            "name": "cause"
          }
        }
      ],
      "memberType": "JavaMethod",
      "modifiers": 1,
      "comment": "Constructs a SQLException object with a given cause. The SQLState is initialized to null and the vendor code is initialized to 0. The reason is initialized to null if cause\u003d\u003dnull or to cause.toString() if cause!\u003dnull.",
      "tagMap": {
        "param": [
          "\u003ccode\u003ecause\u003c/code\u003e - the underlying reason for this \n\u003ccode\u003eSQLException\u003c/code\u003e (which is saved for later retrieval by the \n\u003ccode\u003egetCause()\u003c/code\u003e method); may be null indicating the cause is non-existent or unknown."
        ],
        "since": [
          "1.6"
        ]
      }
    },
    {
      "parameters": [
        {
          "type": {
            "name": "java.lang.String",
            "type": "Class"
          },
          "field": {
            "name": "reason"
          }
        },
        {
          "type": {
            "name": "java.lang.Throwable",
            "type": "Class"
          },
          "field": {
            "name": "cause"
          }
        }
      ],
      "memberType": "JavaMethod",
      "modifiers": 1,
      "comment": "Constructs a SQLException object with a given reason and cause. The SQLState is initialized to null and the vendor code is initialized to 0.",
      "tagMap": {
        "param": [
          "\u003ccode\u003ereason\u003c/code\u003e - a description of the exception.",
          "\u003ccode\u003ecause\u003c/code\u003e - the underlying reason for this \n\u003ccode\u003eSQLException\u003c/code\u003e (which is saved for later retrieval by the \n\u003ccode\u003egetCause()\u003c/code\u003e method); may be null indicating the cause is non-existent or unknown."
        ],
        "since": [
          "1.6"
        ]
      }
    },
    {
      "parameters": [
        {
          "type": {
            "name": "java.lang.String",
            "type": "Class"
          },
          "field": {
            "name": "reason"
          }
        },
        {
          "type": {
            "name": "java.lang.String",
            "type": "Class"
          },
          "field": {
            "name": "sqlState"
          }
        },
        {
          "type": {
            "name": "java.lang.Throwable",
            "type": "Class"
          },
          "field": {
            "name": "cause"
          }
        }
      ],
      "memberType": "JavaMethod",
      "modifiers": 1,
      "comment": "Constructs a SQLException object with a given reason, SQLState and cause. The vendor code is initialized to 0.",
      "tagMap": {
        "param": [
          "\u003ccode\u003ereason\u003c/code\u003e - a description of the exception.",
          "\u003ccode\u003esqlState\u003c/code\u003e - an XOPEN or SQL:2003 code identifying the exception",
          "\u003ccode\u003ecause\u003c/code\u003e - the underlying reason for this \n\u003ccode\u003eSQLException\u003c/code\u003e (which is saved for later retrieval by the \n\u003ccode\u003egetCause()\u003c/code\u003e method); may be null indicating the cause is non-existent or unknown."
        ],
        "since": [
          "1.6"
        ]
      }
    },
    {
      "parameters": [
        {
          "type": {
            "name": "java.lang.String",
            "type": "Class"
          },
          "field": {
            "name": "reason"
          }
        },
        {
          "type": {
            "name": "java.lang.String",
            "type": "Class"
          },
          "field": {
            "name": "sqlState"
          }
        },
        {
          "type": {
            "name": "int",
            "type": "Class"
          },
          "field": {
            "name": "vendorCode"
          }
        },
        {
          "type": {
            "name": "java.lang.Throwable",
            "type": "Class"
          },
          "field": {
            "name": "cause"
          }
        }
      ],
      "memberType": "JavaMethod",
      "modifiers": 1,
      "comment": "Constructs a SQLException object with a given reason, SQLState, vendorCode and cause.",
      "tagMap": {
        "param": [
          "\u003ccode\u003ereason\u003c/code\u003e - a description of the exception",
          "\u003ccode\u003esqlState\u003c/code\u003e - an XOPEN or SQL:2003 code identifying the exception",
          "\u003ccode\u003evendorCode\u003c/code\u003e - a database vendor-specific exception code",
          "\u003ccode\u003ecause\u003c/code\u003e - the underlying reason for this \n\u003ccode\u003eSQLException\u003c/code\u003e (which is saved for later retrieval by the \n\u003ccode\u003egetCause()\u003c/code\u003e method); may be null indicating the cause is non-existent or unknown."
        ],
        "since": [
          "1.6"
        ]
      }
    }
  ],
  "memberType": "JavaClass",
  "modifiers": 1,
  "name": "java.sql.SQLException",
  "comment": "\u003cp\u003eAn exception that provides information on a database access error or other errors. \u003c/p\u003e\n\u003cp\u003eEach \u003ccode\u003eSQLException\u003c/code\u003e provides several kinds of information: \u003c/p\u003e\n\u003cul\u003e \n \u003cli\u003e a string describing the error. This is used as the Java Exception message, available via the method \u003ccode\u003egetMessage\u003c/code\u003e. \u003c/li\u003e\n \u003cli\u003e a \"SQLstate\" string, which follows either the XOPEN SQLstate conventions or the SQL:2003 conventions. The values of the SQLState string are described in the appropriate spec. The \u003ccode\u003eDatabaseMetaData\u003c/code\u003e method \u003ccode\u003egetSQLStateType\u003c/code\u003e can be used to discover whether the driver returns the XOPEN type or the SQL:2003 type. \u003c/li\u003e\n \u003cli\u003e an integer error code that is specific to each vendor. Normally this will be the actual error code returned by the underlying database. \u003c/li\u003e\n \u003cli\u003e a chain to a next Exception. This can be used to provide additional error information. \u003c/li\u003e\n \u003cli\u003e the causal relationship, if any for this \u003ccode\u003eSQLException\u003c/code\u003e. \u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003e\u003c/p\u003e",
  "tagMap": {
    "see": [
      "\u003ca href\u003d\"https://docs.oracle.com/en/java/javase/16/docs/api/serialized-form.html#java.sql.SQLException\"\u003eSerialized Form\u003c/a\u003e"
    ],
    "since": [
      "1.1"
    ]
  }
}